### PR TITLE
Jonmv/async doc v1

### PR DESCRIPTION
**Pull request by jonmv**
@bjorncs or @baldersheim please review.

### CODE REVIEW

File: container-core/src/main/java/com/yahoo/container/core/documentapi/DocumentAccessProvider.java [hunk 1]
```diff
@@ -27,18 +27,18 @@
  *
  * @author jonmv
  */
-public class MessageBusDocumentAccessProvider extends AbstractComponent implements Provider<DocumentAccess> {
+public class DocumentAccessProvider extends AbstractComponent implements Provider<DocumentAccessProvider.LazyWrapper> {
```
Comment (jonmv) [inline container-core/src/main/java/com/yahoo/container/core/documentapi/DocumentAccessProvider.java:32]: This lets us use this in internal classes as well, without fear that the user may have configured their own `DocumentAccess` component.

File: container-core/src/main/java/com/yahoo/container/core/documentapi/DocumentAccessProvider.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package com.yahoo.container.core.documentapi;
```
Comment (bjorncs) [inline container-core/src/main/java/com/yahoo/container/core/documentapi/DocumentAccessProvider.java:1]: Add copyright header to your intellij template :)

File: documentapi/src/main/java/com/yahoo/documentapi/messagebus/MessageBusDocumentAccess.java [hunk 3]
```diff
@@ -60,7 +60,9 @@ public MessageBusDocumentAccess(MessageBusParams params) {
                 bus = new NetworkMessageBus(network, new MessageBus(network, mbusParams));
             }
             else {
-                bus = new RPCMessageBus(mbusParams, params.getRPCNetworkParams(), params.getRoutingConfigId());
+                bus = params.getRPCNetworkParams().getSlobroksConfig() != null && mbusParams.getMessageBusConfig() != null
+                      ? new RPCMessageBus(mbusParams, params.getRPCNetworkParams()) // prefer without self-subscription if config is set
+                      : new RPCMessageBus(mbusParams, params.getRPCNetworkParams(), params.getRoutingConfigId());
```
Comment (bjorncs) [inline documentapi/src/main/java/com/yahoo/documentapi/messagebus/MessageBusDocumentAccess.java:65]: Consider adding warning in log if self-subscription

**Review by bjorncs — APPROVED** [review]

**Comment by bjorncs** [comment]
You'll need to update the reference to provider in config-model

File: container-core/src/main/java/com/yahoo/container/core/documentapi/DocumentAccessProvider.java (hunk 2 above)
Comment (jonmv) [reply container-core/src/main/java/com/yahoo/container/core/documentapi/DocumentAccessProvider.java:2]: Done!

**Review by bjorncs — APPROVED** [review]
👍

**Pull request merged**